File: java/src/test/java/com/wgtwo/auth/Sample.java
package com.wgtwo.auth;

import com.wgtwo.api.v1.sms.SmsProto;
import com.wgtwo.api.v1.sms.SmsProto.SendMessageResponse;
import com.wgtwo.api.v1.sms.SmsProto.SendTextToSubscriberRequest;
import com.wgtwo.api.v1.sms.SmsServiceGrpc;
import com.wgtwo.auth.BearerTokenCallCredentials;
import com.wgtwo.auth.ClientCredentialSource;
import com.wgtwo.auth.WgtwoAuth;
import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;

public class Sample {
    private final String clientId = System.getenv("CLIENT_ID");
    private final String clientSecret = System.getenv("CLIENT_SECRET");
    private final WgtwoAuth wgtwoAuth = WgtwoAuth.builder(clientId, clientSecret).build();
    private final ClientCredentialSource clientCredentialSource = wgtwoAuth.clientCredentialSource("sms.text:send_to_subscriber");
    private final BearerTokenCallCredentials callCredentials = clientCredentialSource.callCredentials();
    private final ManagedChannel channel = ManagedChannelBuilder.forTarget("api.wgtwo.com:443").build();
    private final SmsServiceGrpc.SmsServiceBlockingStub stub = SmsServiceGrpc.newBlockingStub(channel)
            .withCallCredentials(callCredentials);

    public void sendSms() {
        SendTextToSubscriberRequest request = SendTextToSubscriberRequest.newBuilder()
                .setContent("Test")
                .setFromAddress("My Product")
                .setToSubscriber("4799990000")
                .build();
        SendMessageResponse response = stub.sendTextToSubscriber(request);
    }
}

File: java/src/test/kotlin/com/wgtwo/auth/Sample.kt
package com.wgtwo.auth

import com.wgtwo.api.v1.sms.SmsProto.SendTextToSubscriberRequest
import com.wgtwo.api.v1.sms.SmsServiceGrpc
import io.grpc.ManagedChannelBuilder

private val clientId = System.getenv("CLIENT_ID")
private val clientSecret = System.getenv("CLIENT_SECRET")
private val wgtwoAuth = WgtwoAuth.builder(clientId, clientSecret).build()
private val clientCredentialSource = wgtwoAuth.clientCredentials.newTokenSource("sms.text:send_to_subscriber")
private val callCredentials = clientCredentialSource.callCredentials()

fun main() {
    val channel = ManagedChannelBuilder.forTarget("api.wgtwo.com:443").build()
    val stub = SmsServiceGrpc.newBlockingStub(channel)
        .withCallCredentials(callCredentials)

    val request = SendTextToSubscriberRequest.newBuilder().apply {
        content = "Test"
        fromAddress = "My Product"
        toSubscriber = "4799990000"
    }.build()
    val response = stub.sendTextToSubscriber(request)
    println("Sent status: ${response.status}")

    channel.shutdownNow()
}
